### ISSUE TITLE
`columnnames` sometimes returns property names

### ISSUE THREAD

**Issue by ParadaCarleton** [JuliaData/Tables.jl#337]
```
using CSV
data = CSV.load("filepath.csv")
Tables.columnnames(data)
# returns (:filename, :delim, :keywords)--property names--instead of column names
```

**Comment by bkamins** [comment]
Indeed type returned by `CSV.load` seems to need to override the default return-value (or define `getproperty` differently)

**Comment by ParadaCarleton** [comment]
Yep. Besides that, is the default (automatically return properties) a good default, or should it error for types that don't have a defined method?

**Comment by quinnj** [comment]
The default has always been to return `propertynames`.

**Comment by juliohm** [comment]
The linked issue #338 is a good resource. The `Tables.columnnames` trait is only safe in the result of `Tables.columns`. The table itself doesn't store the names of columns. After playing with this design decision for a while, I think it makes sense. Specially because it is hard to introduce fallbacks in Tables.jl with a "catch-all" types dispatch (there is no parent type).

**Comment by quinnj** [comment]
Closing in favor of #338, which tracks clarifying or erroring on this same fallback behavior.
